### DISCUSSION TITLE
vscode.dev: `byteOffset` increased when `FileSystemProvider.writeFile` reads the content of a file

### DISCUSSION THREAD

**Discussion by nturinski** [microsoft/vscode-discussions#469] [Extension Development QnA] [answered]
This only reproduces on `vscode.dev` and seems to be working fine on Vscode desktop.

I'm polyfilling the buffer library with https://github.com/feross/buffer.  Has anyone else run into this? When `FileSystemProvider.readFile,` the content is returned as a `Buffer.from(contents)` which is reading a Unit8Array with 0 `byteOffset`.  However, whenever `FileSystemProvider.writeFile` is called, the `Unit8Array` has the exact same Buffer content, but there's a random `byteOffset` of around 400 which causes the decoding to fail.

It seems like I can resolve this by calling `Buffer.from(uint8Array)` and setting the `byteOffset` to 0, but I was wondering if this was intentional due to running in the browser.

**Comment by olguzzar** [comment]
Bumping this as it's been a while since there was any activity - are you still having this issue?

**Reply by nturinski** [reply, marked as answer]
I still have the workaround set up so I'm not sure. It's fine to close it-- if no one else is noticing this behavior, the workaround works well enough for me.

Was anything done that may have fixed this behavior though?
